web/book/src/internals/modules.md

```diff
@@ -0,0 +1,95 @@
+# Modules
+
+PRQL needs modules that would:
+
+- allow importing declarations from other files,
+- have namespaces for things like `std`,
+- have hierarchical structure so we can represent files in directories,
+- have unambiguous module structure within a project,
+- be able to compile individual files that are a part of a project.
+
+## Basic definition
+
+Module is a namespace that contains declarations.
+
+Modules can be defined with mod keyword and a code block encased in curly
+braces:
+
+```
+mod my_module {
```

Comment:
How come `mod` rather than `module`?

The reason I ask is because elsewhere we've gone for longer names, e.g. `func` rather than `fn`, `aggregate` rather than `agg`, `derive` rather than `drv`, ...

I find those long names a bit of a pain to type sometimes and would probably like all of them shorter. However if we want to be more beginner friendly and explicit then we should probably go for `module` here too in order to be consistent? WDYT?

Reply:
(fwiw this part is likely going away anyway https://github.com/PRQL/prql/pull/2129#issuecomment-1468861960)

Reply:
Why have a code block at all?
It complicated things for the compiler, for users, more to learn, more ways to do one thing, more decide and figure our the preferred way to do things.
I would much prefer there was no `mod` code block and that modules were **always** declared in a self-contained file.

Reply:
I've changed the keyword to `module`, as I agree it is more approachable.